great_expectations/agent/actions/data_assistants/utils.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+from typing import TYPE_CHECKING
+
+from great_expectations.agent.actions.agent_action import (
+    ActionResult,
+)
+from great_expectations.agent.models import (
+    CreatedResource,
+    RunDataAssistantEvent,
+)
+from great_expectations.datasource.fluent import Datasource as FluentDatasource
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+    from great_expectations.core.batch import BatchRequest
+    from great_expectations.rule_based_profiler.data_assistant_result.data_assistant_result import (
+        DataAssistantResult,
+    )
+    from great_expectations.data_context.data_context import CloudDataContext
+
+
+def build_batch_request(
+    context: CloudDataContext,
+    event: RunDataAssistantEvent,
+) -> BatchRequest:
+    datasource = context.get_datasource(datasource_name=event.datasource_name)
+    if not isinstance(datasource, FluentDatasource):
+        raise ValueError(
+            "The RunDataAssistant Action can only be used with a fluent-style datasource."
+        )
+
+    asset = datasource.get_asset(asset_name=event.data_asset_name)
+    try:
+        batch_request = asset.build_batch_request()
+    except ValueError as e:
+        raise ValueError(
+            f"The RunDataAssistant Action for data assistant cannot be used with an in-memory dataframe asset."
+        ) from e
+
+    return batch_request
+
+
+def build_action_result(
+    context: CloudDataContext,
+    event: RunDataAssistantEvent,
+    data_assistant_result: DataAssistantResult,
+) -> ActionResult:
+    # build tz aware timestamp
+    tz = datetime.now().astimezone().tzinfo  # noqa: DTZ005
+    timestamp = datetime.now(tz=tz)
+
+    # ensure we have unique names for created resources
+    expectation_suite_name = f"{event.type} {event.data_asset_name} suite {timestamp}"
+    checkpoint_name = f"{event.type} {event.data_asset_name} checkpoint {timestamp}"
```

Comment:
Nonblocking, but i notice we're now always adding a timestamp - design has consistently asked us to only add timestamp if we have to ([context in slack](https://greatexpectationslabs.slack.com/archives/C056ZF0SXQB/p1686088272369179?thread_ts=1686086296.685339&cid=C056ZF0SXQB)), which is why the previous behavior did what it did.

Reply:
i don't have a strong stance on this, so would suggest to follow up with Trang directly if you feel this is important

Reply:
Okay I'll update this to use the previous logic